### PR TITLE
feat(audit-logs): add filter with context

### PR DESCRIPTION
**Pull request by RemiBonnet**
# What does this PR do?

[> Link to the JIRA ticket
](https://qovery.atlassian.net/jira/software/projects/FRT/boards/23?selectedIssue=FRT-713)

Capture d’écran 2023-07-20 à 14 11 22

- Add filter with context
- Update API doc package
- Refactoring custom filter with outside feature component
- Rename Tools column to Source

---

## PR Checklist

- [ ] This PR introduces breaking change(s) and has been labeled as such
- [ ] This PR introduces new store changes
- [x] I have followed the library pattern i.e `feature`, `ui`, `data`, `utils`
- [x] I made sure the code is type safe (no any)

### CODE REVIEW

**Comment by acarranoqovery** [comment]
[Qovery Preview](https://www.qovery.com)
---

Qovery can create a Preview Environment for this PR.
To trigger its creation, please post a comment with one of the following command.

| Command | Blueprint environment |
|---|---|
| /qovery  preview 28c47145-c8e7-4b9d-8d9e-c65c95b48425 | [staging](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/a64f6cc1-4e34-433b-b8fd-5359716fb01b/environment/28c47145-c8e7-4b9d-8d9e-c65c95b48425/services/general) |
| /qovery  preview {all\|UUID1,UUID2,...} | To preview multiple environments |

---

This comment has been generated from [Qovery](https://qovery.com) AI :robot:.
Below, a word from its wisdom :
> There are two hard problems in computer science: naming things, cache invalidation and off-by-1 errors

File: libs/pages/events/src/lib/feature/custom-filter-feature/custom-filter-feature.tsx [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+  )
+
+  const { data: eventsTargetsData, isLoading: isLoadingEventsTargetsData } = useFetchEventTargets(
+    organizationId,
+    queryParams,
+    displayEventTargets
+  )
+
+  useEffect(() => {
+    const newQueryParams: EventQueryParams = extractEventQueryParams(location.pathname + location.search)
```
Comment (ctjhoa) [inline libs/pages/events/src/lib/feature/custom-filter-feature/custom-filter-feature.tsx:40]: Not very efficient queryParams management.
[should be tackled in technical task]

File: libs/pages/events/src/lib/feature/custom-filter-feature/custom-filter-feature.tsx [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+    if (newQueryParams.fromTimestamp && newQueryParams.toTimestamp)
+      setTimestamps([
+        new Date(parseInt(newQueryParams.fromTimestamp, 10) * 1000),
```
Comment (ctjhoa) [inline libs/pages/events/src/lib/feature/custom-filter-feature/custom-filter-feature.tsx:44]: Not very efficient queryParams management.
[should be tackled in technical task]

File: libs/pages/events/src/lib/feature/custom-filter-feature/custom-filter-feature.tsx [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+        new Date(parseInt(newQueryParams.toTimestamp, 10) * 1000),
+      ])
+  }, [location])
+
+  const onChangeTimestamp = (startDate: Date, endDate: Date) => {
+    setSearchParams((prev) => {
+      prev.delete('continueToken')
+      prev.delete('stepBackToken')
+      prev.set('fromTimestamp', convertDatetoTimestamp(startDate.toString()).toString())
```
Comment (ctjhoa) [inline libs/pages/events/src/lib/feature/custom-filter-feature/custom-filter-feature.tsx:53]: Poor date conversion.
[should be tackled in technical task]

**Review by ctjhoa — APPROVED** [review]

**Pull request merged**